src/dev/performance/run_scalability_cli.ts

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { run } from '@kbn/dev-cli-runner';
+import { REPO_ROOT } from '@kbn/utils';
+import fs from 'fs';
+import path from 'path';
+
+run(
+  async ({ log, flagsReader, procRunner }) => {
+    async function runScalabilityJourney(journeyConfigPath: string) {
+      // Pass in a clean APM environment, so that FTR can later
+      // set it's own values.
+      const cleanApmEnv = {
+        ELASTIC_APM_ACTIVE: undefined,
+        ELASTIC_APM_BREAKDOWN_METRICS: undefined,
+        ELASTIC_APM_CONTEXT_PROPAGATION_ONLY: undefined,
+        ELASTIC_APM_CAPTURE_SPAN_STACK_TRACES: undefined,
+        ELASTIC_APM_ENVIRONMENT: undefined,
+        ELASTIC_APM_GLOBAL_LABELS: undefined,
+        ELASTIC_APM_MAX_QUEUE_SIZE: undefined,
+        ELASTIC_APM_METRICS_INTERVAL: undefined,
+        ELASTIC_APM_SERVER_URL: undefined,
+        ELASTIC_APM_SECRET_TOKEN: undefined,
+        ELASTIC_APM_TRANSACTION_SAMPLE_RATE: undefined,
+      };
+
+      await procRunner.run('scalability-tests', {
+        cmd: 'node',
+        args: [
+          'scripts/functional_tests',
+          ['--config', 'x-pack/test/scalability/config.ts'],
+          ['--kibana-install-dir', kibanaInstallDir],
+          '--debug',
+          '--logToFile',
+          '--bail',
+        ].flat(),
+        cwd: REPO_ROOT,
+        wait: true,
+        env: {
+          ...cleanApmEnv,
+          SCALABILITY_JOURNEY_PATH: journeyConfigPath, // Input file for Gatling test runner
+          KIBANA_DIR: REPO_ROOT, // Gatling test runner use it to find kbn/es archives
+        },
+      });
+    }
+
+    const kibanaInstallDir = flagsReader.requiredPath('kibana-install-dir');
+    const journeyConfigPath = flagsReader.requiredPath('journey-config-path');
+    const journeys = fs.statSync(journeyConfigPath).isDirectory()
+      ? fs
+          .readdirSync(journeyConfigPath)
+          .filter((fileName) => path.extname(fileName) === '.json')
+          .map((fileName) => path.resolve(journeyConfigPath, fileName))
+      : [journeyConfigPath];
+
+    log.info(`Found ${journeys.length} journeys to run:\n${JSON.stringify(journeys)}`);
+
+    const failedJourneys = [];
+
+    for (const journey of journeys) {
+      try {
+        process.stdout.write(`--- Running scalability journey: ${journey}\n`);
```

Comment:
any reason to not use the logger here?

Reply:
I'm trying to use [Buildkite collapsable output](https://buildkite.com/docs/pipelines/managing-log-output#collapsing-output) here, so that each test has its own section.

<img width="1116" alt="image" src="https://user-images.githubusercontent.com/10977896/205942203-73a3ee7c-9de7-4c04-b1bb-769a3350a448.png">

to make it work log line should starts with `---`
e.g. `echo "--- Running scalability journey: ${journey}"`

I don't think I can make it work with logger, or do I miss something?

Reply:
ah got it, I missed that. :+1: